indico/modules/events/registration/google_wallet.py

```diff
@@ -0,0 +1,265 @@
+                        'value': self.event.address
+                    }
+                }
+            },
+            'logo': {
+                'sourceUri': {
+                    'uri': logo_url
+                }
+            },
+            'textModulesData': [
+                {
+                    'header': 'Date',
+                    'language': 'en-US',
+                    'body': make_format_event_date_func(self.event.category)(self.event),
+                    'id': 'datefield'
+                },
+            ],
+            'classTemplateInfo': {
+                'cardTemplateOverride': {
+                    'cardRowTemplateInfos': [{
+                        'oneItem': {
+                            'item': {
+                                'firstValue': {
+                                    'fields': [{
+                                        'fieldPath': "class.textModulesData['datefield']"
+                                    }]
+                                }
+                            },
+                        }
+                    }, {
+                        'twoItems': {
+                            'startItem': {
+                                'firstValue': {
+                                    'fields': [{
+                                        'fieldPath': "object.textModulesData['namefield']"
+                                    }]
+                                }
+                            },
+                            'endItem': {
+                                'firstValue': {
+                                    'fields': [{
+                                        'fieldPath': "object.textModulesData['emailfield']"
+                                    }]
+                                }
+                            },
+                        }
+                    }]
+                }
+            }
+        }
+        if not dict_template['venue']['name']['defaultValue']['value']:
+            dict_template.pop('venue')
+        if self.event.has_logo and self.event.is_public:  # Event logo MUST be reachable via URL as anon
+            dict_template['heroImage'] = {
+                'sourceUri': {
+                    'uri': self.event.external_logo_url
+                }
+            }
+        signals.event.google_wallet_class_template_created.send(self.event, data=dict_template)
+        return dict_template
+
+    def create_class(self, issuer_id: str, class_suffix: str, update: bool = False) -> str:
+        """Create/update a class."""
+        # Check if the class exists
+        response = self.http_client.get(url=f'{self.class_url}/{issuer_id}.{class_suffix}')
+
+        if response.status_code == 200:
+            # Class already exists
+            if update:
+                updated_class = self.create_class_template(issuer_id, class_suffix)
+                response = self.http_client.put(url=f'{self.class_url}/{issuer_id}.{class_suffix}', json=updated_class)
```

Comment:
Do we always need to do an update? Or only if certain things changed? Since you get the class first (ie you can compare things), it seems a bit unnecessary to always send an update even if nothing changed.

Reply:
ah, didn't notice it's behind another flag. fine then, i thought you're always sending an update just in case something changed